### ISSUE TITLE
Phase-fix idiom has a NaN VJP at zero: jnp.where does not short-circuit the 0/0 branch (4 sites, incl. the default phase gauge)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#789]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

Found while fixing the same class of defect in `_normalise_rdm` (#748 / PR #788); filing separately rather than folding it into that branch.

## The idiom

Four sites normalise a complex number to a pure phase like this:

```python
phases = jnp.where(jnp.abs(signs) > 0, signs / jnp.abs(signs), 1.0)
```

- `src/tenax/algorithms/_ad_primitives.py:104`
- `src/tenax/algorithms/_ctm_root_implicit_asym.py:344`
- `src/tenax/algorithms/_ctm_root_implicit_symmetric.py:393`
- `src/tenax/algorithms/_ctm_tensor_projector_2x2.py:53`

The `jnp.where` makes the **value** correct at `signs = 0`. It does not make the **gradient** correct, because `jnp.where` does not short-circuit a VJP: the unselected branch still computes `0/0 = NaN` and contributes `0 * NaN = NaN`.

## Reproduction

```python
import jax, jax.numpy as jnp
jax.config.update("jax_enable_x64", True)

def phase_fix(signs):
    return jnp.where(jnp.abs(signs) > 0, signs / jnp.abs(signs), 1.0)

f = lambda x: jnp.sum(jnp.abs(phase_fix(x)) ** 2).real

jax.grad(f)(jnp.array([1.0 + 1j, 0.0 + 0j, 2.0 + 0j]))
# -> [2.22e-16-2.22e-16j,  nan+nanj,  0.+0.j]
```

Note the failure shape: **only the zero element is poisoned**. The gradient looks healthy everywhere else, so this does not announce itself — it silently NaNs one column of whatever it feeds.

## Why it is reachable

`signs` is a diagonal of `R` (QR) or a reference element. On the block-sparse path an exactly-zero diagonal entry is ordinary, not exotic — an empty or rank-deficient charge sector produces one. `_ctm_tensor_projector_2x2.py:53` sits on the **phase gauge, which is the production default**.

This is plausibly related to the 9 NaN-AD failures bisected to PR #406 (`_gauge_fixed_svd` backward on degenerate SVs) — same "NaN only in the degenerate/zero subspace" signature. Worth checking whether it is the same root cause before treating them as two bugs.

## The fix, and the house style that already gets it right

Fence the *argument* of the non-differentiable op, not just the result — the `S_rsqrt` idiom already used at `_ctm_projector.py:805`, `_ctm_compiled_moves.py:102`, `_tensor_utils.py:92` and elsewhere does this correctly:

```python
mask   = S > tol
S_safe = jnp.where(mask, S, 1.0)          # sqrt never sees 0
S_rsqrt = jnp.where(mask, 1.0 / jnp.sqrt(S_safe), 0.0)
```

Verified NaN-free at zero. So the correct pattern is already in the codebase; these four sites just predate or missed it.

## Suggested scope

1. Fix the four sites with the fenced-argument form.
2. Add a gradient-at-zero guard per site, mutation-verified (a value-only test passes against the broken code — that is exactly how this survived).
3. Check the #406 NaN-AD cluster against this root cause.
